core/src/main/java/org/infinispan/commands/RemoveCacheCommand.java

```diff
@@ -23,31 +18,19 @@
    public static final byte COMMAND_ID = 18;
 
    private EmbeddedCacheManager cacheManager;
-   private GlobalComponentRegistry registry;
-   private PersistenceManager persistenceManager;
-   private CacheJmxRegistration cacheJmxRegistration;
 
    private RemoveCacheCommand() {
       super(null); // For command id uniqueness test
    }
 
-   public RemoveCacheCommand(String cacheName, EmbeddedCacheManager cacheManager,
-         GlobalComponentRegistry registry, PersistenceManager persistenceManager,
-         CacheJmxRegistration cacheJmxRegistration) {
+   public RemoveCacheCommand(String cacheName, EmbeddedCacheManager cacheManager) {
       super(cacheName);
       this.cacheManager = cacheManager;
-      this.registry = registry;
-      this.persistenceManager = persistenceManager;
-      this.cacheJmxRegistration = cacheJmxRegistration;
    }
 
    @Override
    public Object perform(InvocationContext ctx) throws Throwable {
-      persistenceManager.setClearOnStop(true);
-      cacheJmxRegistration.setUnregisterCacheMBean(true);
-      Cache<Object, Object> cache = cacheManager.getCache(cacheName);
-      cache.stop();
-      registry.removeCache(cacheName);
+      ((DefaultCacheManager) cacheManager).removeCacheLocally(cacheName);
```

Comment:
This will most likely blow up when running within Wildfly since Wildfly does not use DefaulCacheManager implementation directly AFAIK. They implement a delegating cache manager AFAIK.

I really don't see the need for a `removeCacheLocally` method at all. Can't you create a private method in this command, get the component registry from the cache manager, which can be injected, and in the private method do everything you do in removeCacheLocally? If the method needs to be accessible from multiple places, add it to a util class as static method passing in all the parameters needed?